application/themehandler.class.inc.php

```diff
@@ -273,9 +294,278 @@ public static function ComputeSignature($aThemeParameters, $aImportsPaths)
 			$sFile = static::FindStylesheetFile($sStylesheet, $aImportsPaths);
 			$aSignature['stylesheets'][$key] = md5_file($sFile);
 		}
+		foreach ($aIncludedImages as $sImage)
+		{
+			if (is_file($sImage))
+			{
+				$aSignature['images'][$sImage] = md5_file($sImage);
+			}
+		}
+
 		return json_encode($aSignature);
 	}
 
+	const IMAGE_EXTENSIONS = array('png', 'gif', 'jpg', 'jpeg');
+
+	/**
+	 * Search for images referenced in stylesheet files
+	 * @param $aThemeParametersVariables
+	 * @param $aStylesheetFile
+	 * @param $sThemeFolderPath : used as relative paths to find css images
+	 *
+	 * @return array
+	 */
+	public static function GetIncludedImages($aThemeParametersVariables, $aStylesheetFile, $sThemeFolderPath)
+	{
+		$aCompleteUrls = array();
+		$aToCompleteUrls = array();
+		$aMissingVariables = array();
+		$aFoundVariables = array();
+		$aMap = array(
+			'aCompleteUrls' => $aCompleteUrls,
+			'aToCompleteUrls' => $aToCompleteUrls,
+			'aMissingVariables' => $aMissingVariables,
+			'aFoundVariables' => $aFoundVariables,
+		);
+
+		foreach ($aStylesheetFile as $sStylesheetFile)
+		{
+			$aRes = self::GetAllUrlFromScss($aThemeParametersVariables, $sStylesheetFile);
+			foreach($aMap as $key => /** array */$aVal)
+			{
+				if (array_key_exists($key, $aMap))
+				{
+					$aMap[$key]=array_merge($aVal, $aRes[$key]);
+				}
+			}
+		}
+
+		$aMap = ThemeHandler::ResolveUncompleteUrlsFromScss($aMap, $aThemeParametersVariables, $aStylesheetFile);
```

Comment:
If you are in the class itself, rather use "static"?

```suggestion
		$aMap = static::ResolveUncompleteUrlsFromScss($aMap, $aThemeParametersVariables, $aStylesheetFile);
```